design-proposals/export-import/volume-export.md

```diff
@@ -0,0 +1,161 @@
+
+#### dir
+
+The http endpoint is a golang `http.FileServer` serving up the root of the filesystem.
+
+### Export a VirtualMachine
+
+``` yaml
+apiVersion: export.kubevirt.io/v1alpha1
+kind: VirtualMachineExport
+metadata:
+    name: export-vm1
+spec:
+    source:
+        apiGroup: kubevirt.io
+        kind: VirtualMachine
+        name: vm1
+```
+
+### Export a VirtualMachineSnapshot
+
+``` yaml
+apiVersion: export.kubevirt.io/v1alpha1
+kind: VirtualMachineExport
+metadata:
+    name: export-snap1
+spec:
+    source:
+        apiGroup: snapshot.kubevirt.io
+        kind: VirtualMachineSnapshot
+        name: snap1
+```
+
+### VirtualMachineExport Status
+
+``` yaml
+apiVersion: export.kubevirt.io/v1alpha1
+kind: VirtualMachineExport
+metadata:
+    namespace: ns1
+    name: export-vm1
+spec:
+    ...
+status:
+    phase: Ready # Pending|Ready|Terminated
+    token: 8fmf94kf
+    links:
+      internal:
+        cert: <base64 encoded cert>
+        volumes:
+        - name: kubevirt-volume
+          - format: raw
+            url: https://virt-import.kubevirt/ns1/export-vm1/volumes/kubevirt-volume/raw?token=8fmf94kf
+          - format: archive
+            url: https://virt-import.kubevirt/ns1/export-vm1/volumes/kubevirt-volume/archive?token=8fmf94kf
+        - name: archive-volume
+          - format: dir
+            url: https://virt-import.kubevirt/ns1/export-vm1/volumes/archive-volume/dir?token=8fmf94kf
+          - format: archive
+            url: https://virt-import.kubevirt/ns1/export-vm1/volumes/archive-volume/archive?token=8fmf94kf
+      external:
+        cert: <base64 encoded cert>
+        volumes:
+        - name: kubevirt-volume
+          - format: raw
+            url: https://virt-import.kubevirt.my-cluster/ns1/export-vm1/volumes/kubevirt-volume/raw?token=8fmf94kf
+          - format: archive
+            url: https://virt-import.kubevirt.my-cluster/ns1/export-vm1/volumes/kubevirt-volume/archive?token=8fmf94kf
+        - name: archive-volume
+          - format: dir
+            url: https://virt-import.kubevirt.my-cluster/ns1/export-vm1/volumes/archive-volume/dir?token=8fmf94kf
+          - format: archive
+            url: https://virt-import.kubevirt.my-cluster/ns1/export-vm1/volumes/archive-volume/archive?token=8fmf94kf
+    conditions:
+    - type: Ready
+      status: True
+      ...
+      reason: Ready to export Virtual Machine
+```
+
+* Cluster must have Ingress/Route support for `external` urls and certificate to be set
+
+## virtctl export
+
+virtctl will be extended to include an `export` command that will create/delete a `VirtualMachineExport`
+
+```
+virtctl export create --vm=vm1 vm1-export
+```
+
+```
+virtctl export create --snapshot=snap1 snap1-export
+```
+
+```
+virtctl export delete snap1-export
+```
+
+Once a `VirtualMachineExport` is created, `virtctl export` can download volume archives
+
+```
+virtctl export download vm1-export --volume=volume1 --output volume1.tar.gz
```

Comment:
What about download a bundle of all the volumes of the vm?

Reply:
yeah, I'll add that, the trick will be to do it in a way that preserves sparseness somehow.  Should be possible